From af528051ee3a6066f3e19637cdb7c0037a1be74e Mon Sep 17 00:00:00 2001
From: Billy Laws <blaws05@gmail.com>
Date: Thu, 16 Jan 2025 20:13:21 +0000
Subject: [PATCH 1060/1504] include: Use inline assembly for TEB access with
 clang.

Avoids typ conflicts when WINE_NO_LONG_TYPES is defined.
---
 include/winnt.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/winnt.h b/include/winnt.h
index 52a48e4249b..6c0afa341af 100644
--- a/include/winnt.h
+++ b/include/winnt.h
@@ -2452,7 +2452,7 @@ NTSYSAPI struct _TEB * WINAPI NtCurrentTeb(void) __attribute__((pure));
 # else
 NTSYSAPI struct _TEB * WINAPI NtCurrentTeb(void);
 # endif
-#elif defined(__i386__) && defined(__GNUC__)
+#elif defined(__i386__) && (defined(__GNUC__) || defined(__clang__))
 static FORCEINLINE struct _TEB * WINAPI NtCurrentTeb(void)
 {
     struct _TEB *teb;
-- 
2.49.0

